crates/bevy_ecs/src/system/builder.rs

```diff
@@ -710,6 +711,36 @@ unsafe impl<'w, 's, T: FnOnce(&mut FilteredResourcesMutBuilder)>
     }
 }
 
+/// A [`SystemParamBuilder`] for an [`Option`].
+#[derive(Clone)]
+pub struct OptionBuilder<T>(T);
+
+// SAFETY: `WhenBuilder<B>` builds a state that is valid for `P`, and any state valid for `P` is valid for `When<P>`
```

Comment:
the docs should say `OptionBuilder<B>`, right?

Reply:
Oh, yup!  I guess you can tell where I copied it from :).